pkg/cli/admin/upgrade/recommend/alerts.go

```diff
@@ -0,0 +1,185 @@
+package recommend
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/oc/pkg/cli/admin/inspectalerts"
+	"github.com/openshift/oc/pkg/cli/admin/upgrade/status"
+)
+
+func (o *options) alerts(ctx context.Context) ([]metav1.Condition, error) {
+	var alertsBytes []byte
+	if o.mockData.alertsPath != "" {
+		if len(o.mockData.alerts) == 0 {
+			return []metav1.Condition{{
+				Type:    "recommended/Alert",
+				Status:  metav1.ConditionTrue,
+				Reason:  "NoTestData",
+				Message: fmt.Sprintf("This --mock-clusterversion run did not have alert data available at %v", o.mockData.alertsPath),
+			}}, nil
+		}
+		alertsBytes = o.mockData.alerts
+	} else {
+		client, err := routev1client.NewForConfig(o.RESTConfig)
+		if err != nil {
+			return nil, err
+		}
+		routeGetter := func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
+			return client.Routes(namespace).Get(ctx, name, opts)
+		}
+
+		alertsBytes, err = inspectalerts.GetAlerts(ctx, routeGetter, o.RESTConfig.BearerToken)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var alertData status.AlertData
+	err := json.Unmarshal(alertsBytes, &alertData)
+	if err != nil {
+		return nil, fmt.Errorf("parsing alerts: %w", err)
+	}
+
+	var conditions []metav1.Condition
+	i := 0
+	haveCritical := false
+	havePodDisruptionBudget := false
+	havePullWaiting := false
+	haveNodes := false
+	for _, alert := range alertData.Data.Alerts {
+		var alertName string
```

Comment:
At _some_ point I'll probably want to refactor this so the single alert processing logic is a unit-testable method well covered by unit test (what output we expect for a specific alerts) but for now it's fine.